### PR TITLE
feat: Add Card page

### LINKED ISSUE #27: Create Card component page

### PR DESCRIPTION
**Pull request by acdvorak**
Fixes #27

screenshot of card page

### CODE REVIEW

File: src/images/ic_card_24px.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...24"><defs><path id="a" d="M-240.34-61.17c-18.22 0-33 15.15-33 33.84 0 14.95 9.45 27.63 22.57 32.11 1.65.31 2.25-.74 2.25-1.63 0-.8-.03-2.93-.04-5.75-9.18 2.04-11.12-4.54-11.12-4.54-1.5-3.91-3.66-4.95-3.66-4.95-3-2.1.23-2.06.23-2.06 3.31.24 5.05 3.49 5.05 3.49 2.94 5.17 7.72 3.68 9.6 2.81.3-2.19 1.15-3.68 2.09-4.52-7.33-.85-15.03-3.76-15.03-16.72 0-3.69 1.29-6.71 3.4-9.08-.34-.86-1.47-4.3.32-8.95 0 0 2.77-.91 9.08 3.47 2.63-.75 5.46-1.13 8.26-1.14 2.8.01 5.63.39 8.26 1.14 6.3-4.38 9.07-3.47 9.07-3.47 1.8 4.66.67 8.1.33 8.95 2.11 2.37 3.39 5.39 3.39 9.08 0 13-7.72 15.86-15.07 16.7 1.19 1.04 2.24 3.11 2.24 6.27 0 4.52-.04 8.17-.04 9.28 0 .91.59 1.96 2.27 1.63 13.1-4.48 22.55-17.16 22.55-32.1.01-18.71-14.77-33.86-33-33.86"/></defs><clipPath id="b"><use xlink:href="#a" overflow="visible"/></clipPath><g clip-path="url(#b)"><defs><path id="c" d="M-1121.33-2515.17h1680v3369h-1680z"/></defs><clipPath id="d"><use xlink:href="#c" overflow="visible"/></clipPath><path clip-path="url(#d)" d="M-278.33-66.17h76v76h-76z"/></g><path d="M20 4v16H4V4h16m2-2H2v20h20V2z"/><path d="M5 12.54h6.46V19H5zm7.54 0H19V19h-6.46zM5 5h6.46v6.46H5zm7.54 0H19v6.46h-6.46z"/><path fill="none" d="M0 0h24v24H0z"/></svg>
```
Comment (moog16) [inline src/images/ic_card_24px.svg]: This one can be deleted

File: src/styles/ComponentPage.scss [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
   align-items: center;
   justify-content: center;
-  height: 360px;
   min-height: 360px;
+  padding: 24px;
   background-color: rgba(0, 0, 0, .05);
 }
 
-.sidebar-active {
+.sidebar-active.mdc-list-item {
```
Comment (moog16) [inline src/styles/ComponentPage.scss]: I remember someone saying not to use `.mdc-foo` selectors in the scss. I think just to be safe use a `.demo-` class.

File: src/ComponentPage.js [hunk 3]
```diff
@@ -46,18 +57,17 @@ class ComponentPage extends Component {
...
         <div className='hero'>
           {this.props.hero}
         </div>
-        <h2 className='demo-title mdc-typography--title'>Resources</h2>
-        {this.renderResource('Material Design Guidelines', `${imagePath}/ic_material_design_24px.svg`,
-          this.props.designLink)}
-        {this.renderResource('Documentation', `${imagePath}/ic_drive_document_24px.svg`, this.props.docsLink)}
-        {this.renderResource('Source Code', `${imagePath}/ic_code_24px.svg`, this.props.sourceLink)}
+        <h2 className='demo-title mdc-typography--headline6'>Resources</h2>
+        {this.renderResource('Material Design Guidelines', materialLogo, this.props.designLink)}
```
Comment (moog16) [inline src/ComponentPage.js]: These 3 image links should be reverted back

File: src/CardPage.js [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+              </div>
+            </div>
+          </div>
+
+          <div className='mdc-card mdc-card--outlined demo-card'>
+            <div className='demo-card-article-group-heading mdc-typography--subtitle2'>Headlines</div>
+
+            <hr className='mdc-list-divider'/>
+
+            <div className='demo-card-article mdc-ripple-surface' tabIndex='0' ref={this.initRipple}>
```
Comment (moog16) [inline src/CardPage.js]: Here are 3 of the same node structures. You could make another component here and no need to have a constructor in the parent component.

```
class CardArticle extends Component {
  ripple = null;
  
  componentWillUnmount() {
    this.ripple.destroy();
  }
  render() {
    return (
      <div
        className='demo-card-article mdc-ripple-surface'
        tabIndex='0' 
        ref={(surfaceEl) => this.ripple = new MDCRipple(surfaceEl)}
      >
        <h2 className='demo-card-article__title mdc-typography--headline5'>{this.props.title}</h2>
        <p className='demo-card-article__snippet mdc-typography--body2'>
          {this.props.subtitle}
        </p>
      </div>
    );
  }
}
```

File: src/CardPage.js [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+                  <div className='demo-card__music-title mdc-typography--headline5'>Rozes</div>
+                  <div className='demo-card__music-artist mdc-typography--body2'>Under the Grave</div>
+                  <div className='demo-card__music-year mdc-typography--body2'>(2016)</div>
+                </div>
+              </div>
+            </div>
+            <hr className='mdc-list-divider'/>
+            <div className='mdc-card__actions'>
+              <div className='mdc-card__action-buttons demo-card__action-buttons--text-only'>Rate this album</div>
+              <div className='mdc-card__action-icons'>
```
Comment (moog16) [inline src/CardPage.js]: These `<i>`s can also become a self contained component so there's not as much repetition.

```
class CardActionIcon extends Component {
  ripple = null;
  
  componentWillUnmount() {
    this.ripple.destroy();
  }
  render() {
    return (
      <i className='material-icons demo-card__action-icon--star mdc-ripple-surface'
         tabIndex='0'
         role='button'
         title={this.props.title}
         data-mdc-ripple-is-unbounded
         ref={(surfaceEl) => this.ripple = new MDCRipple(surfaceEl)}>
         star_border
      </i>
    );
  }
}

```


Then the markup would become:

```
<div className='mdc-card__action-icons'>
  <CardActionIcon title='1 star'/>
  <CardActionIcon title='2 star'/>
  <CardActionIcon title='3 star'/>
  <CardActionIcon title='4 star'/>
  <CardActionIcon title='5 star'/>
</div>

File: src/CardPage.js [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+  componentWillUnmount() {
+    this.iconToggles.forEach((iconToggle) => iconToggle.destroy());
+    this.ripples.forEach((ripple) => ripple.destroy());
+  }
+
+  render() {
+    return (
+      <div>
+        <section className='demo-card-collection'>
+          <div className='mdc-card demo-card demo-card--photo'>
```
Comment (moog16) [inline src/CardPage.js]: It would be good to break up each card into either its own card or renderMethod (eg. `renderPhotoCard()`). Then it could just look like:

```
<section className='demo-card-collection'>
  <CardPhoto />
  <CardMusic />
  <CardHeadlines />
</section>
```

File: src/styles/ComponentPage.scss (hunk 2 above)
Comment (kfranqueiro) [reply src/styles/ComponentPage.scss]: Why was a 2nd class added here? I can't tell what this needs to override that it wasn't already overriding.

Comment (moog16) [reply src/styles/ComponentPage.scss]: Looks like the font-weight is being overwritten

screen shot 2018-05-01 at 15 58 02

File: public/index.html [hunk 7]
```diff
@@ -12,7 +12,7 @@
     <link rel="shortcut icon" href="%PUBLIC_URL%/mdc_web_48dp.png">
     <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
     <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Mono">
-    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500">
+    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,600,700">
```
Comment (lynnjepsen) [inline public/index.html]: do we need this?

File: src/ButtonPage.js [hunk 8]
```diff
@@ -67,7 +67,7 @@ class ButtonDemos extends Component {
   renderButtonVariant(title, variantClass) {
     return (
       <div>
-        <h3 className='mdc-typography--subheading2'>{title}</h3>
+        <h3 className='mdc-typography--subtitle1'>{title}</h3>
```
Comment (lynnjepsen) [inline src/ButtonPage.js]: why are you changing Button in this Card PR?

File: src/CardPage.js [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+              <h2 className='demo-card__title mdc-typography--headline6'>Our Changing Planet</h2>
+              <h3 className='demo-card__subtitle mdc-typography--subtitle2'>by Kurt Wagner</h3>
+            </div>
+            <div className='demo-card__secondary mdc-typography--body2'>
+              Visit ten places on our planet that are undergoing the biggest changes today.
+            </div>
+          </div>
+          <div className='mdc-card__actions'>
+            <div className='mdc-card__action-buttons'>
+              <button className='mdc-button mdc-card__action mdc-card__action--button'>Read</button>
```
Comment (lynnjepsen) [inline src/CardPage.js]: add the ripple to the buttons

File: src/CardPage.js [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+              <button className='mdc-button mdc-card__action mdc-card__action--button'>Bookmark</button>
+            </div>
+            <div className='mdc-card__action-icons'>
+              <i className='mdc-icon-toggle material-icons mdc-card__action mdc-card__action--icon'
```
Comment (lynnjepsen) [inline src/CardPage.js]: add the ripple to these icons too

File: src/ComponentPage.js [hunk 11]
```diff
@@ -12,7 +12,7 @@ class ComponentPage extends Component {
         href={`${process.env.PUBLIC_URL}${link.url}${pageExt}`}
         key={index}
         role='listitem'
-        className={`mdc-list-item ${link.active ? 'sidebar-active' : ''}`}>
+        className={`mdc-list-item mdc-typography--body1 ${link.active ? 'sidebar-active' : ''}`}>
```
Comment (lynnjepsen) [inline src/ComponentPage.js]: split this into another PR

File: src/ComponentPage.js [hunk 12]
```diff
@@ -62,18 +66,17 @@ class ComponentPage extends Component {
   renderDemoWrapper() {
     return(
       <section className='demo-wrapper mdc-layout-grid__cell mdc-layout-grid__cell--span-10'>
-        <h1 className='mdc-typography--headline'>{this.props.title}</h1>
+        <h1 className='mdc-typography--headline5'>{this.props.title}</h1>
```
Comment (lynnjepsen) [inline src/ComponentPage.js]: And split this into another PR...all of this

File: package.json [hunk 13]
```diff
@@ -6,10 +6,13 @@
     "@material/button": "^0.35.0",
     "@material/checkbox": "^0.35.0",
     "@material/fab": "^0.35.0",
+    "@material/card": "^0.35.0",
+    "@material/icon-toggle": "^0.35.0",
     "@material/image-list": "^0.35.0",
     "@material/layout-grid": "^0.34.0",
     "@material/list": "^0.35.0",
     "@material/ripple": "^0.35.0",
+    "@material/rtl": "^0.35.0",
```
Comment (lynnjepsen) [inline package.json]: do you need RTL?

File: src/styles/CardPage.scss [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+  width: 350px;
+  margin: 48px 0;
+}
+
+.demo-card--hero {
+  margin: 0;
+}
+
+//
+// Book card
```
Comment (lynnjepsen) [inline src/styles/CardPage.scss]: Book card?
